rust_snuba/src/consumer.rs

```diff
@@ -122,9 +123,24 @@ pub fn consumer_impl(
         Some(consumer_config.raw_topic.broker_config),
     );
 
-    let consumer = Box::new(KafkaConsumer::new(config));
     let logical_topic_name = consumer_config.raw_topic.logical_topic_name;
 
+    let consumer_state = Arc::new(Mutex::new(ConsumerState::new(Box::new(
```

Comment:
Instead of having to manually wrap this in `Arc`, `Mutex`, `ConsumerState` and `Box`, how about you make a default trait method `ProcessingStrategyFactory::into_consumer_state` which does this for you.